### PR TITLE
feat(simulator_launch): replace with tier4_simulator_launch

### PR DESCRIPTION
**Pull request by kminoda**
## PR Type

- New Feature

## Related Links
- Localization: https://github.com/tier4/autoware_launch/pull/651
- Perception: https://github.com/tier4/autoware_launch/pull/681

## Description
Remove simulator_launch and replace with tier4_simulator_launch.

Tested with psim

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [x] Code follows [coding guidelines][coding-guidelines]
- [x] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] Code follows [coding guidelines][coding-guidelines]
- [ ] (Optional) Unit tests have been written for new behavior
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets
- [ ] Write [release notes][release-notes]

## CI Checks

- **Build and test for PR**: Required to pass before the merge.
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[coding-guidelines]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/1194394777/T4
[release-notes]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/563774416
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

File: autoware_launch/launch/planning_simulator.launch.xml [hunk 1]
```diff
@@ -60,7 +60,7 @@
     <let name="launch_dummy_vehicle" value="true" unless="$(var scenario_simulation)"/>
     <let name="launch_dummy_localization" value="true"/>
 
-    <include file="$(find-pkg-share simulator_launch)/launch/simulator.launch.xml">
+    <include file="$(find-pkg-share autoware_launch)/launch/components/tier4_simulator_component.launch.xml">
```
Comment (yukkysaito) [inline autoware_launch/launch/planning_simulator.launch.xml:63]: There is also a digital twin simulation and a rosbag replay simulation, so the file name is better to clear that it is for a planning simulation.

Comment (kminoda) [reply autoware_launch/launch/planning_simulator.launch.xml:63]: Thank you!
Do you mean something like `/launch/components/tier4_planning_simulator_component.launch.xml`?
Maybe we also should modify the name `tier4_simulator_launch` in autoware.universe?

Comment (yukkysaito) [reply autoware_launch/launch/planning_simulator.launch.xml:63]: > Do you mean something like /launch/components/tier4_planning_simulator_component.launch.xml?
> Maybe we also should modify the name tier4_simulator_launch in autoware.universe?

Yes :+1:

Comment (kminoda) [reply autoware_launch/launch/planning_simulator.launch.xml:63]: I see, thank you. 

But changing the name of `tier4_simulator_component.launch.xml` and `tier4_simulator_launch` sounds out of scope of this PR.

How about we first go on with `tier4_simulator_component.launch.xml` for this PR, and discuss about renaming the launch files in another PR (or maybe internal / public discussion)?

Comment (yukkysaito) [reply autoware_launch/launch/planning_simulator.launch.xml:63]: OK :+1: 
I'll resolve this conversation

**Review by yukkysaito — APPROVED** [review]
LGTM

**Pull request merged**